### PR TITLE
Add file permissions to Zend/Log/Writter/Stream

### PR DESCRIPTION
**Pull request by raultabacu**
https://github.com/zendframework/zend-log/issues/11

### CODE REVIEW

File: src/Writer/Stream.php [hunk 1]
```diff
@@ -85,6 +86,12 @@ public function __construct($streamOrUrl, $mode = null, $logSeparator = null)
                     $mode
                 ), 0, $error);
             }
+            if (!is_null($filePermissions) && !chmod($streamOrUrl, $filePermissions)) {
```
Comment (zf2timo) [inline src/Writer/Stream.php]: `null !== $filePermissions`

**Comment by zf2timo** [comment]
Why did you removed the Unit Test. I think we need them for this change.

**Comment by raultabacu** [comment]
Because chmod throws an exception when working with 'php://memory'

**Comment by zf2timo** [comment]
You could mock the Filesystem:
https://phpunit.de/manual/current/en/test-doubles.html#test-doubles.mocking-the-filesystem

**Comment by raultabacu** [comment]
vfsStream is not installed on the test servers. Is there a different path for it, or do I need to add it manually ?

**Comment by zf2timo** [comment]
You could install it via composer.

**Comment by weierophinney** [comment]
This partially duplicates #7, but does so via an explicit constructor argument, vs a named option.

I've merged it, with the following changes:
- added vfsStream as a development requirement in `composer.json`.
- updated usage of vfsStream, as the proposed usage was creating files in the filesystem; the new usage does not.
- merged the changes with those already merged for #7 to allow both options for setting the file permissions.

Merged to develop for release with 2.7.0.

**Pull request closed without merging**